File: app/src/main/java/com/sami/rippel/WallpaperApplication.java
package com.sami.rippel;


import com.sami.rippel.core.DaggerApplicationComponent;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

/**
 * Created by yassin baccour on 15/05/2016.
 */

public class WallpaperApplication extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerApplicationComponent.factory().create(this);
    }

}

File: app/src/main/java/com/yassin/wallpaper/WallpaperApplication.kt
package com.yassin.wallpaper

import android.app.Application
import com.yassin.wallpaper.appInitializers.AppInitializers
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject

/**
 * Created by yassin baccour on 15/05/2016.
 */
@HiltAndroidApp
class WallpaperApplication : Application() {

    @Inject
    lateinit var initializers: AppInitializers

    override fun onCreate() {
        super.onCreate()
        initializers.init(this)
    }
}
